lib/ansible/utils/collection_loader/_collection_finder.py

```diff
@@ -355,7 +369,8 @@ def find_spec(self, fullname, path, target=None):
         if loader is None:
             return None
 
-        spec = spec_from_loader(fullname, loader)
+        origin = getattr(loader, 'path', None)
```

Comment:
This change, plus the related change in `_get_subpackage_search_paths` allows `find_spec` to return a `ModuleSpec` containing an `origin` for redirected builtins.  This will simplify logic when trying to get a path for a module/plugin via `ansible_collections` import.